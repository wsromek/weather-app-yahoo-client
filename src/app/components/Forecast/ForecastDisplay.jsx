import React from 'react';
import TodayForecast from './TodayForecast';
import UpcomingDaysForecast from './UpcomingDaysForecast';

function ForecastDisplay({ forecast: { units, item, astronomy, atmosphere, wind, location } }) {
  if (Object.keys(arguments[0].forecast).length === 0) {
    return null;
  }

  const dailyForecasts = item.forecast;
  const currentConditions = item.condition;

  const todayConditions = {
    units,
    currentConditions,
    atmosphere,
    astronomy,
    wind,
  };

  return (
    <section className='forecast'>
      <h2>Current condition in {location.city}, {location.region}</h2>

      <TodayForecast
        conditions={todayConditions}
      />
      <UpcomingDaysForecast
        dailyForecasts={dailyForecasts}
        temperatureUnit={units.temperature}
      />
    </section>
  );

}

export default ForecastDisplay;

