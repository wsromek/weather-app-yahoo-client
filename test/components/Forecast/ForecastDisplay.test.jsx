import React from 'react';
import renderer from 'react-test-renderer';
import forecastFixture from '../../fixtures/forecast';
import ForecastDisplay from '../../../src/app/components/Forecast/ForecastDisplay';

describe('ForecastDisplay component', () => {
  it('renders correctly', () => {
    const present = renderer.create(
      <ForecastDisplay forecast={forecastFixture} />,
    ).toJSON();

    const absent = renderer.create(
      <ForecastDisplay forecast={{}} />,
    ).toJSON();

    expect(present).toMatchSnapshot();
    expect(absent).toMatchSnapshot();
  });
});
