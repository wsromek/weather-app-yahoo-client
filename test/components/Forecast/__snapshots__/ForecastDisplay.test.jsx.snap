// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDisplay component renders correctly 1`] = `
<section
  className="forecast"
>
  <h2>
    Current condition in 
    San Francisco
    , 
    CA
  </h2>
  <div
    className="jumbotron"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-4"
        >
          <div
            className="tile"
          >
            <h3>
              Wind
            </h3>
            <div>
              Speed: 
              7
               
              mph
            </div>
            <div>
              Chill: 
              54° F
            </div>
            <div>
              Direction: 
              245
              °
            </div>
            <div
              className="arrow"
              style={
                Object {
                  "transform": "rotate(245deg)",
                }
              }
            >
              ↑
            </div>
          </div>
        </div>
        <div
          className="col-sm-4"
        >
          <div
            className="tile"
          >
            <h3>
              Conditions
            </h3>
            <div>
              Temp: 
              54° F
            </div>
            <img
              alt="condition_image"
              src="http://l.yimg.com/a/i/us/we/52/31.gif"
            />
            <div>
              <div>
                Sunrise: 
                6:15 am
              </div>
              <div>
                Sunset: 
                7:59 pm
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-sm-4"
        >
          <div
            className="tile"
          >
            <h3>
              Atmosphere
            </h3>
            <div>
              Humidity: 
              77
              %
            </div>
            <div>
              Pressure: 
              1017
               
              in
            </div>
            <div>
              Visibility: 
              16.1
               
              mi
               
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h2>
      Upcoming 
      10
       days
    </h2>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-4"
        >
          <div
            className="daily-forecast-wrapper"
          >
            <div
              className="forecast-date"
            >
              30 Apr 2017
            </div>
            <img
              alt="condition_image"
              src="http://l.yimg.com/a/i/us/we/52/32.gif"
            />
            <div
              className="forecast-temp-range"
            >
              Temp: 
              54
              ° 
              - 
              68
              ° 
            </div>
          </div>
        </div>
        <div
          className="col-sm-4"
        >
          <div
            className="daily-forecast-wrapper"
          >
            <div
              className="forecast-date"
            >
              01 May 2017
            </div>
            <img
              alt="condition_image"
              src="http://l.yimg.com/a/i/us/we/52/32.gif"
            />
            <div
              className="forecast-temp-range"
            >
              Temp: 
              56
              ° 
              - 
              71
              ° 
            </div>
          </div>
        </div>
        <div
          className="col-sm-4"
        >
          <div
            className="daily-forecast-wrapper"
          >
            <div
              className="forecast-date"
            >
              02 May 2017
            </div>
            <img
              alt="condition_image"
              src="http://l.yimg.com/a/i/us/we/52/34.gif"
            />
            <div
              className="forecast-temp-range"
            >
              Temp: 
              57
              ° 
              - 
              70
              ° 
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-4"
        >
          <div
            className="daily-forecast-wrapper"
          >
            <div
              className="forecast-date"
            >
              03 May 2017
            </div>
            <img
              alt="condition_image"
              src="http://l.yimg.com/a/i/us/we/52/34.gif"
            />
            <div
              className="forecast-temp-range"
            >
              Temp: 
              59
              ° 
              - 
              72
              ° 
            </div>
          </div>
        </div>
        <div
          className="col-sm-4"
        >
          <div
            className="daily-forecast-wrapper"
          >
            <div
              className="forecast-date"
            >
              04 May 2017
            </div>
            <img
              alt="condition_image"
              src="http://l.yimg.com/a/i/us/we/52/30.gif"
            />
            <div
              className="forecast-temp-range"
            >
              Temp: 
              57
              ° 
              - 
              67
              ° 
            </div>
          </div>
        </div>
        <div
          className="col-sm-4"
        >
          <div
            className="daily-forecast-wrapper"
          >
            <div
              className="forecast-date"
            >
              05 May 2017
            </div>
            <img
              alt="condition_image"
              src="http://l.yimg.com/a/i/us/we/52/28.gif"
            />
            <div
              className="forecast-temp-range"
            >
              Temp: 
              56
              ° 
              - 
              63
              ° 
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-4"
        >
          <div
            className="daily-forecast-wrapper"
          >
            <div
              className="forecast-date"
            >
              06 May 2017
            </div>
            <img
              alt="condition_image"
              src="http://l.yimg.com/a/i/us/we/52/30.gif"
            />
            <div
              className="forecast-temp-range"
            >
              Temp: 
              54
              ° 
              - 
              63
              ° 
            </div>
          </div>
        </div>
        <div
          className="col-sm-4"
        >
          <div
            className="daily-forecast-wrapper"
          >
            <div
              className="forecast-date"
            >
              07 May 2017
            </div>
            <img
              alt="condition_image"
              src="http://l.yimg.com/a/i/us/we/52/34.gif"
            />
            <div
              className="forecast-temp-range"
            >
              Temp: 
              55
              ° 
              - 
              63
              ° 
            </div>
          </div>
        </div>
        <div
          className="col-sm-4"
        >
          <div
            className="daily-forecast-wrapper"
          >
            <div
              className="forecast-date"
            >
              08 May 2017
            </div>
            <img
              alt="condition_image"
              src="http://l.yimg.com/a/i/us/we/52/30.gif"
            />
            <div
              className="forecast-temp-range"
            >
              Temp: 
              55
              ° 
              - 
              64
              ° 
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-4"
        >
          <div
            className="daily-forecast-wrapper"
          >
            <div
              className="forecast-date"
            >
              09 May 2017
            </div>
            <img
              alt="condition_image"
              src="http://l.yimg.com/a/i/us/we/52/30.gif"
            />
            <div
              className="forecast-temp-range"
            >
              Temp: 
              56
              ° 
              - 
              63
              ° 
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`ForecastDisplay component renders correctly 2`] = `null`;
